gemrb/core/GameScript/GSUtils.cpp

```diff
@@ -1603,18 +1603,14 @@ static int ParseIntParam(const char *&src, const char *&str)
 	}
 	if (*str=='*') { //there may be an IDS table
 		str++;
-		char idsTabName[9];
-		char *cur = idsTabName;
-		const char *end = idsTabName + 9 - 1;
-		while (*str != ',' && *str != ')') {
-			// limit IDS file length to 8 characters
-			 if (cur != end) {
-				*cur = *str;
-				++cur;
-			}
+		ResRef idsTabName;
+		char *cur = idsTabName.begin();
+		const char *end = idsTabName.bufend();
+		while (cur != end && *str != ',' && *str != ')') {
+			*cur = *str;
+			++cur;
 			++str;
 		}
```

Comment:
This can be further simplified by using `FindNotOf` instead of the loop.

Reply:
`FindNotOf` wont work on a c string tho. Needs further refactoring to get there.